File: src/main/java/com/manifestors/shinrai/client/command/commands/ToggleCommand.java
package com.manifestors.shinrai.client.command.commands;

import com.manifestors.shinrai.client.Shinrai;
import com.manifestors.shinrai.client.command.Command;

public class ToggleCommand extends Command {

    public ToggleCommand() {
        super("toggle", "Toggles a module by name.", "module", "t");
    }

    @Override
    public boolean onCommandExecuted(String[] args) {
        if (args.length < 2)
            return false;
        var moduleName = args[1];
        var module = Shinrai.INSTANCE.getModuleManager().getModuleByName(moduleName);

        if (module != null) {
            module.toggleModule();
            Shinrai.INSTANCE.addChatMessage(module.getName() + " has been " + (module.isEnabled() ? "enabled" : "disabled") + ".");
            return true;
        } else
            Shinrai.INSTANCE.addChatMessage("Module not found.");

        return false;
    }
}

File: src/main/kotlin/com/manifestors/shinrai/client/command/commands/ToggleCommand.kt
package com.manifestors.shinrai.client.command.commands

import com.manifestors.shinrai.client.Shinrai.addChatMessage
import com.manifestors.shinrai.client.Shinrai.moduleManager
import com.manifestors.shinrai.client.command.Command

class ToggleCommand : Command(
    commandName = "toggle",
    commandDescription = "Toggles a module by name.",
    validParameters = "module",
    "t"
) {

    override fun onCommandExecuted(args: Array<String>): Boolean {
        if (args.size < 2) return false

        val moduleName = args[1]
        moduleManager.getModuleByName(moduleName)?.let {
           it.toggleModule()
           addChatMessage("${it.name} has been ${if (it.enabled) "enabled" else "disabled"}.")
        } ?: sendNotFound("Module '$moduleName'")

        return true
    }
}
